src/components/elements/input/ClientImageUploadDialog.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Button,
+  ButtonProps,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogProps,
+  DialogTitle,
+  Grid,
+  Typography,
+} from '@mui/material';
+import { omit } from 'lodash-es';
+import React, { useCallback, useState } from 'react';
+
+import { ClientCardImageElement } from '../ClientCard';
+import Uploader from '../upload/Uploader';
+
+import {
+  useGetClientImageQuery,
+  useUpdateClientImageMutation,
+} from '@/types/gqlTypes';
+
+export type ClientImageUploadDialogProps = {
+  clientId: string;
+} & DialogProps;
+
+const ClientImageUploadDialog: React.FC<ClientImageUploadDialogProps> = ({
+  clientId,
+  onClose,
+  ...props
+}) => {
+  const { data: { client } = {}, loading } = useGetClientImageQuery({
+    variables: { id: clientId },
+  });
+  const [mutate, { loading: updating }] = useUpdateClientImageMutation();
+
+  const [newPhotoSrc, setNewPhotoSrc] = useState<string | undefined>();
+  const [newBlobId, setNewBlobId] = useState<string | undefined>();
+
+  const handleClose = useCallback<NonNullable<ButtonProps['onClick']>>(
+    (e) => {
+      setNewPhotoSrc(undefined);
+      setNewBlobId(undefined);
+      if (onClose) onClose(e, 'escapeKeyDown');
+    },
+    [onClose]
+  );
+
+  const handleSave = useCallback<NonNullable<ButtonProps['onClick']>>(
+    (e) => {
+      if (newBlobId)
+        mutate({ variables: { clientId, imageBlobId: newBlobId } }).then(() =>
+          handleClose(e)
+        );
+    },
+    [newBlobId, clientId, mutate, handleClose]
+  );
+
+  if (!loading && !client) return null;
+
+  return (
+    <Dialog {...omit(props, 'children')} onClose={onClose}>
+      <DialogTitle>Upload Client Photo</DialogTitle>
+      <DialogContent>
+        {loading ? (
+          <CircularProgress />
+        ) : (
+          client && (
+            <Grid container spacing={2} justifyContent='space-around'>
+              <Grid item>
+                <Typography>Current Photo</Typography>
+                {client.image && <ClientCardImageElement client={client} />}
+              </Grid>
+              <Grid item>
+                <Typography>New Photo</Typography>
+                {client.image && <ClientCardImageElement url={newPhotoSrc} />}
+              </Grid>
+              <Grid item xs={12}>
+                <Uploader
```

Comment:
Not everyone is familiar with the drag and drop. There should also be an "Choose File" button that pulls up the standard file selection tool, both here and in the form. Thanks!

Reply:
Fixed